commands/commands.go

```diff
@@ -315,15 +315,49 @@ func requireInRepo() {
 // that it not be bare. If it is bare (or the state of the repository could not
 // be determined), this function will terminate the program.
 func requireWorkingCopy() {
+	if cfg.LocalWorkingDir() == "" {
+		Print("This operation must be run in a work tree.")
+		os.Exit(128)
+	}
+}
+
+func setupRepository() {
+	requireInRepo()
 	bare, err := git.IsBare()
 	if err != nil {
 		ExitWithError(errors.Wrap(
 			err, "fatal: could not determine bareness"))
 	}
 
-	if bare {
-		Print("This operation must be run in a work tree.")
-		os.Exit(128)
+	if !bare {
+		changeToWorkingCopy()
+	}
+}
+
+func setupWorkingCopy() {
+	requireInRepo()
+	requireWorkingCopy()
+	changeToWorkingCopy()
+}
+
+func changeToWorkingCopy() {
+	workingDir := cfg.LocalWorkingDir()
+	cwd, err := os.Getwd()
```

Comment:
IIUC, the path which we get for the current working directory will be compared down below (after running `filepath.EvalSymlinks()` on it) to the output from `cfg.LocalWorkingDir()` which I believe [ultimately](https://github.com/git-lfs/git-lfs/blob/68a8f8a49d834c91c9e9d39bc53c6fc065122dbd/config/config.go#L368) comes from `git.GitAndRootDirs()`which as of 23fd6c5973b69694776e630e11eda48bfe1f838c is now always going to call the new `tools.CanonicalizePath()` which will call `TranslateCygwinPath()`.

So in order to also run `TranslateCygwinPath()` on the output from `os.Getwd()`, perhaps this should instead call `tools.Getwd()`?

Reply:
Good point.  Updated.